src/Symfony/Component/DependencyInjection/Loader/PhpFileLoader.php

```diff
@@ -57,17 +57,40 @@ public function load(mixed $resource, ?string $type = null): mixed
         $this->setCurrentDir(\dirname($path));
         $this->container->fileExists($path);
 
+        // Force load ContainerConfigurator to make env(), param() etc available.
+        class_exists(ContainerConfigurator::class);
+
         // the closure forbids access to the private scope in the included file
         $load = \Closure::bind(function ($path, $env) use ($container, $loader, $resource, $type) {
             return include $path;
         }, $this, ProtectedPhpFileLoader::class);
 
         try {
-            $callback = $load($path, $this->env);
+            $result = $load($path, $this->env);
 
-            if (\is_object($callback) && \is_callable($callback)) {
-                $this->executeCallback($callback, new ContainerConfigurator($this->container, $this, $this->instanceof, $path, $resource, $this->env), $path);
+            if (\is_object($result) && \is_callable($result)) {
+                $result = $this->executeCallback($result, new ContainerConfigurator($this->container, $this, $this->instanceof, $path, $resource, $this->env), $path);
             }
+            if ($result instanceof ConfigBuilderInterface) {
+                $this->loadExtensionConfig($result->getExtensionAlias(), ContainerConfigurator::processValue($result->toArray()));
+            } elseif (is_iterable($result)) {
+                foreach ($result as $key => $config) {
+                    if ($config instanceof ConfigBuilderInterface) {
+                        if (\is_string($key) && $config->getExtensionAlias() !== $key) {
+                            throw new InvalidArgumentException(\sprintf('The extension alias "%s" of the "%s" config builder does not match the key "%s" in file "%s".', $config->getExtensionAlias(), get_debug_type($config), $key, $path));
+                        }
+                        $this->loadExtensionConfig($config->getExtensionAlias(), ContainerConfigurator::processValue($config->toArray()));
+                    } elseif (!\is_string($key) || !\is_array($config)) {
+                        throw new InvalidArgumentException(\sprintf('The configuration returned in file "%s" must yield only string-keyed arrays or ConfigBuilderInterface values.', $path));
+                    } else {
+                        $this->loadExtensionConfig($key, ContainerConfigurator::processValue($config));
+                    }
+                }
+            } elseif (\is_string($result) || \is_object($result)) {
```

Comment:
what about other kinds of results (a number for instance) ?

Reply:
we have to ignore it because int or bool are what include returns when the included file doesn't "return"

Reply:
the `include` return value is `1|false`. We could be stricter by only silently accepting `1`, `false` and `null` (used by `executeCallback` when it skips the callable).
Note that even `false` is not fully relevant there, as it would happen only if we fail to include the file, and this case should be considered as an error anyway.